### ISSUE TITLE
Facing issues while setting up OAuth2 with Microsoft Exchange

### ISSUE THREAD

**Issue by d78ui98** [ecederstrand/exchangelib#1332]
**Describe the bug**
Hello team,

I’m encountering issues while setting up OAuth2 with Microsoft Exchange.

I’m using a Google Chronicle SOAR's Exchange integration created instance with the required credentials (client ID, client secret, tenant ID). During the process of generating the refresh token, the integration uses the get authorization action to produce a URL. As per the documentation, this URL should provide a code field to manually generate the refresh token after login. However, instead of the expected URL, I’m receiving error message :
`https://login.microsoftonline.com/<tenant-id>/Consent/Set`

Could you please advise on how to resolve this and obtain the correct authorization URL?

Also here is the documentation we are following - https://cloud.google.com/chronicle/docs/soar/marketplace-integrations/exchange

**Comment by ecederstrand** [comment]
Can you please post the current code you're using to try to connect?

**Comment by d78ui98** [comment]
Hello Erik,

Thanks for your comment. I managed to resolve the issue. The code was fine, but the problem was with the redirect URL and permissions.

Initially, I followed documentation that suggested `http://localhost` as the redirect URL, but I found another source recommending `https://login.microsoftonline.com/common/oauth2/nativeclient`. After updating the redirect URL in the Microsoft Entra app and correct application permissions, everything worked perfectly.

Expected URL: `https://login.microsoftonline.com/common/oauth2/nativeclient?code=<authcode>`

We can close this issue now. I appreciate your time!
